operators/pkg/controller/elasticsearch/certificates/transport/pod_secret.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package transport
+
+import (
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/name"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// LabelAssociatedPod is a label key that indicates the resource is supposed to have a named associated pod
+	LabelAssociatedPod = "transport.certificates.elasticsearch.k8s.elastic.co/associated-pod"
+
+	// LabelSecretUsage is a label key that specifies what the secret is used for
+	LabelSecretUsage = "transport.certificates.elasticsearch.k8s.elastic.co/secret-usage"
+	// LabelSecretUsageTransportCertificates is the LabelSecretUsage value used for transport certificates
+	LabelSecretUsageTransportCertificates = "transport-certificates"
+
+	// LabelTransportCertificateType is a label key indicating what the transport-certificates secret is used for
+	LabelTransportCertificateType = "transport.certificates.elasticsearch.k8s.elastic.co/certificate-type"
+	// LabelTransportCertificateTypeElasticsearchAll is the LabelTransportCertificateType value used for Elasticsearch
+	LabelTransportCertificateTypeElasticsearchAll = "elasticsearch.all"
```

Comment:
Not sure I follow the intention behind this labelling scheme

Reply:
This is not a new labelling scheme, just moved code.

I think this should go away completely. "nodecerts" was aiming to cover potentially multiple kinds of certs (for different use cases etc) with a truly complicated labelling scheme that since it wasn't extended, didn't work out as intended. It seems much simpler to just use a naming convention here, which is far simpler and covers our existing needs.

That being said, it felt a little out of scope for this specific PR, so I left it in instead of ripping it out.

Reply:
It feels more natural to cover this through https://github.com/elastic/cloud-on-k8s/issues/722